src/Views/Components/Stream.php

```diff
@@ -9,22 +9,34 @@
 
 class Stream extends Component
 {
-    /** @var string|Model|array */
-    public $target;
+    public string|Model|array|null $target;
 
-    public string $action;
+    public ?string $action;
+
+    public ?string $targets;
 
     /**
      * Create a new component instance.
      *
-     * @param string|Model|array $target The DOM ID string, a model to generate the DOM ID for, or an array to be passed to the `dom_id` function.
-     * @param string $action One of the seven Turbo Stream actions: "append", "prepend", "before", "after", "replace", "update", or "remove".
-     * @return void
+     * @param string|Model|array|null $target The DOM ID string, a model to generate the DOM ID for, or an array to be passed to the `dom_id` function.
+     * @param ?string $action One of the seven Turbo Stream actions: "append", "prepend", "before", "after", "replace", "update", or "remove".
+     * @param string|null $targets The CSS selector to apply the action to multiple targets
      */
-    public function __construct($target, string $action)
+    public function __construct(string|Model|array|null $target = null, ?string $action = null, ?string $targets = null)
```

Comment:
This was the part of the PR we struggled with most.

Options we considered:

* Making a fully separate Stream class with support for just targets (breaks compatibility with Turbo handbook)
* Adding an optional `isMultipleTargets` bool that changes the behaviour of `$target` (breaks compatibility with Turbo handbook since you don't have a `targets` property)

In the end, we went with this option that adds default values for `$target` and `$action`, and has a rather awkward parameter order.
As you can tell, we're not 100% happy about this, and we're hoping to have a decision around the best way to implement this.

Were we to implement this with a clean sheet, we'd probably go `__construct(string $action, string|Model|array|null $target = null, ?string $targets = null)`.

Reply:
Yeah, I think I prefer what you described at the end, moving the required parameter so its before the optional ones. I've already tackled that [here](https://github.com/tonysm/turbo-laravel/pull/87)

Reply:
I'm not so worried about the BC change. Except maybe for the queued job (`BroadcastAction`), which is where we need to be careful when changing attributes, signatures and stuff (since users of the package may have serialized jobs in the queue when a deployment with the updated package version goes live - I'm gonna test for this before tagging a release)